File: BACKEND-APPLICATION/src/main/java/com/application/entities/Reservation.java
package com.application.entities;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Setter
@Getter
@Table(name = "reservations")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String checking;
    private String checkout;
    @ManyToOne
    private Client client;
    @ManyToOne
    private Room room;
    boolean isApproved;

    public Reservation() { }

}

File: FRONTEND-APPLICATION/src/app/models/reservation.ts
import { Room } from "./room";
import { User } from "./user";

export class Reservation {

  public id?: number;
  public checking:any;
  public checkout:any;
  public room: Room;
  public client: User;
  public approved?:boolean;


  constructor() {
    this.approved = false;
    this.room = new Room();
    this.client = new User();
    this.checking = new Date();
    this.checkout = new Date();

  }

}